File: DEVBGJAVADEMO/DEVBGJAVADEMO/src/test/java/hooks/UiHooks.java
package hooks;

import io.cucumber.java.After;
import io.cucumber.java.Before;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;

public class UiHooks {
    public static WebDriver driver;
    public static String createdEmail;
    public static String createdPassword;
    public static String BASE_UI_URL = "https://demoqa.com";

    @Before(order = 100)
    public void startBrowser(){
        ChromeOptions opts = new ChromeOptions();
        driver = new ChromeDriver(opts);
        driver.manage().window().maximize();
    }

    @After
    public void quitBrowser(){
        if(driver != null){
            driver.quit();
        }
    }
}

File: DEVBGDEMO/DEVBGDEMO/Hooks/UiHooks.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using Reqnroll;

namespace DEVBGDEMO.Hooks
{
    [Binding]
    public class UiHooks
    {
        public static IWebDriver? Driver { get; private set; }
        public static string? CreatedEmail { get; set; }
        public static string? CreatedPassword { get; set; }
        public static string? Products { get; set; }
        public static string? ProductNumber { get; set; }
        public static string? ProductName { get; set; }
        public static string BASE_UI_URL = "https://demoqa.com";

        [BeforeScenario(Order = 100)]
        public void StartBrowser()
        {
            var options = new ChromeOptions();
            options.AddArgument("--start-maximized");
            Driver = new ChromeDriver(options);
            Driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(2);
            Driver.Manage().Window.Maximize();
        }

        [AfterScenario]
        public void QuitBrowser()
        { 
            Driver?.Quit(); 
        }
    }
}
